src/WebJobs.Script/Extensions/FunctionMetadataExtensions.cs

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using Microsoft.Azure.WebJobs.Script.Description;
+
+namespace Microsoft.Azure.WebJobs.Script.Extensions
+{
+    public static class FunctionMetadataExtensions
+    {
+        public static bool IsSimpleHttpTriggerFunction(this FunctionMetadata functionMetadata)
+        {
```

Comment:
If we're calling this from every invocation, we should probably be caching the result. Either with a new property in FunctionMetadata or some some dictionary map here. We don't really need to go through this for every invocation, do we?

Reply:
We do need this per invocation. You are right adding a new property on FunctionMetadata makes sense. Will update.